### ISSUE TITLE
epel-next-9 broken, missing epel-release package

### ISSUE THREAD

**Issue by praiskup** [rpm-software-management/mock#790]
Chroot installation fails with:
```
...
Extra Packages for Enterprise Linux 9 - Next - x86_64                                                                                                                                                       
148 kB/s |  36 kB     00:00    
No match for argument: epel-release
No match for argument: epel-rpm-macros
No match for argument: fedpkg-minimal
No match for argument: epel-next-release
```

@carlwgeorge seems like we should override `chroot_setup_cmd` in epel-next-9.tpl, not just
append `epel-next-release` to it?

**Comment by carlwgeorge** [comment]
No, the initial packages will have the gpg key in epel-release and the epel-next repo config in epel-next-release, with epel-next-release requiring epel-release.  See [this build](https://koji.fedoraproject.org/koji/buildinfo?buildID=1845105).  Once epel9 is ready we'll add the epel repo config to epel-release by enabling [this bcond](https://src.fedoraproject.org/rpms/epel-release/blob/epel9-next/f/epel-release.spec#_1).  I'm expecting to be able to get those packages into the repo later this week.

**Comment by praiskup** [comment]
I meant, shouldn't we do it "for now"?  Because it clearly makes `mock -r epel-next-9-x86_64` fail. And at the same time we try to make the configuration work [by a temporary hack](https://github.com/rpm-software-management/mock/blob/16db6b71c14dbf90145e4e32f5b4e9f098fe72ee/mock-core-configs/etc/mock/templates/epel-9.tpl#L1-L4) which makes no sense (if we still fail).

Also, I don't fully understand what we need `epel-release` and `fedpkg-minimal` for, don't you have more info about that? It was originally added in [RHBZ#1764445](https://bugzilla.redhat.com/1764445) but I'm not really sure what is it for. I also [asked](https://pagure.io/releng/issue/10369) the releng folks, as the Koji situation motivated us to add those originally in daa2de45c193300ba335145b1283316f84422e96.

Yes, `epel-rpm-macros` are needed (and till we have them, builds will fail I guess).

The last one needed - `epel-next-release` - what do we need it for?

> No, the initial packages will have the gpg key in epel-release

Mock uses keys from distribution-gpg-keys?

**Comment by carlwgeorge** [comment]
You raise good points.  I set up epel9-next to replicate what we had with epel8-next, which was based on what was set up for epel8.  It does indeed seem that epel-release and epel-next-release are not needed.

epel-rpm-macros and fedpkg-minimal are still needed.  I can send a PR to modify the configs.

**Comment by tdawson** [comment]
Do we need the epel gpg keys for anything during a mock build and/or setup?
Those are found in the epel-release package.

**Comment by carlwgeorge** [comment]
The mock configs use the key from distribution-gpg-keys, e.g.:

https://github.com/rpm-software-management/mock/blob/e43db17bbfbcd4f05f5f22a0ba3f6a5440bed261/mock-core-configs/etc/mock/templates/epel-next-9.tpl#L10

With the key from distribution-gpg-keys and the repo definition from mock-core-configs, I believe @praiskup is correct that we don't need to install epel-release or epel-next-release.

**Comment by praiskup** [comment]
Fixed in 6c23518

**Comment by xsuchy** [comment]
This issue motivated me to document how we handle GPG keys: https://github.com/rpm-software-management/mock/pull/795
